contracts/vaults/STETHVault/RibbonThetaSTETHVault.sol

```diff
@@ -0,0 +1,356 @@
+        auctionDuration = _auctionDuration;
+    }
+
+    /************************************************
+     *  SETTERS
+     ***********************************************/
+
+    /**
+     * @notice Sets the new discount on premiums for options we are selling
+     * @param newPremiumDiscount is the premium discount
+     */
+    function setPremiumDiscount(uint16 newPremiumDiscount) external onlyOwner {
+        require(
+            newPremiumDiscount > 0 && newPremiumDiscount < 1000,
+            "Invalid discount"
+        );
+
+        premiumDiscount = newPremiumDiscount;
+    }
+
+    /**
+     * @notice Sets the new auction duration
+     * @param newAuctionDuration is the auction duration
+     */
+    function setAuctionDuration(uint256 newAuctionDuration) external onlyOwner {
+        require(newAuctionDuration >= 1 hours, "!newAuctionDuration");
+
+        auctionDuration = newAuctionDuration;
+    }
+
+    /**
+     * @notice Sets the new strike selection or options premium pricer contract
+     * @param newContract is the address of the new strike selection or options premium pricer contract
+     * @param isStrikeSelection is whether we are setting the strike selection contract
+     */
+    function setStrikeSelectionOrPricer(
+        address newContract,
+        bool isStrikeSelection
+    ) external onlyOwner {
+        require(newContract != address(0), "!newContract");
+        if (isStrikeSelection) {
+            strikeSelection = newContract;
+        } else {
+            optionsPremiumPricer = newContract;
+        }
+    }
+
+    /**
+     * @notice Withdraws the assets on the vault using the outstanding `DepositReceipt.amount`
+     * @param amount is the amount to withdraw in `asset`
+     * @param minETHOut is the min amount of `asset` to recieve for the swapped amount of steth in crv pool
+     */
+    function withdrawInstantly(uint256 amount, uint256 minETHOut)
+        external
+        nonReentrant
+    {
+        Vault.DepositReceipt storage depositReceipt =
+            depositReceipts[msg.sender];
+
+        uint256 currentRound = vaultState.round;
+
+        require(amount > 0, "!amount");
+        require(depositReceipt.round == currentRound, "Invalid round");
+
+        uint256 receiptAmount = depositReceipt.amount;
+        require(receiptAmount >= amount, "Exceed amount");
+
+        // Subtraction underflow checks already ensure it is smaller than uint104
+        depositReceipt.amount = uint104(receiptAmount.sub(amount));
+        vaultState.totalPending = uint128(
+            uint256(vaultState.totalPending).sub(amount)
+        );
+
+        // Subtraction underflow checks already ensure it is smaller than uint104
+        depositReceipt.amount = uint104(uint256(receiptAmount).sub(amount));
+
+        emit InstantWithdraw(msg.sender, amount, currentRound);
+
+        // Unwrap may incur curve pool slippage
+        uint256 amountETHOut =
+            VaultLifecycleSTETH.unwrapYieldToken(
```

Comment:
Why would you need to do `unwrapYieldToken` when it's `withdrawInstantly`? Shouldn't the token still be WETH since it's not rolled yet?

Reply:
firstly, it would be eth (no weth deposits). 

secondly, you are assuming I am depositing eth. what if I am depositing steth? we don't allow for withdrawYieldToken() so we have to do this check. _most likely_ in the future there will be enough other eth deposits in the same round such that we wont actually have to unwrap. so unwrapYieldToken will just exit out after a few lines. but in the off chance we need to unwrap, we need this.